client/cli/src/logger.rs

```diff
@@ -0,0 +1,270 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use ansi_term::Colour;
+use flexi_logger::{
+    DeferredNow, Duplicate, LogSpecBuilder,
+    LogSpecification, LogTarget, Logger, Criterion, Naming, Cleanup, Age,
+};
+use lazy_static::lazy_static;
+use regex::Regex;
+use std::path::PathBuf;
+use structopt::{
+	StructOpt,
+};
+use crate::error::{Error, Result};
+
+type IoResult = std::result::Result<(), std::io::Error>;
+
+/// Default size used for rotation. Corresponds to 100 MiB.
+const DEFAULT_ROTATION_SIZE: u64 = 104_857_600;
+
+/// Options for log rotation.
+#[derive(Debug, StructOpt)]
+pub struct LogRotationOpt {
+	/// Specify the path of the directory which will contain the log files.
+	/// Defaults to rotating logs once they reach 100 MiB.
```

Comment:
Why is this 100 MiB value mentioned here and not for `log-size`?

Reply:
Hmm because there is no "default" for `log-size`. Meaning if one uses `--log-size`, he is expected to provide a value. 
- `substrate --log-directory /tmp/my_logs` -> will log files in provided path, rotating every time a log reached 100 MiB.
- `substrate --log-directory /tmp/my_logs --log-size` errors because log-size does not have a default
- `substrate --log-directory /tmp/my_logs --log-size 1073741824`will rotate every time a log file reaches 1Gib.

Let me know it you think I should still add it to `--log-size` ! :)